app/controllers/concerns/foreman/controller/registration.rb

```diff
@@ -18,6 +18,12 @@ def global_registration_vars
     location = Location.authorized(:view_locations).find(params['location_id']) if params['location_id'].present?
     host_group = Hostgroup.authorized(:view_hostgroups).find(params['hostgroup_id']) if params["hostgroup_id"].present?
     operatingsystem = Operatingsystem.authorized(:view_operatingsystems).find(params['operatingsystem_id']) if params["operatingsystem_id"].present?
+
+    if params["repo"].present?
+      repo_data = {}
+      repo_data[params['repo']] = params['repo_gpg_key_url'] || ''
+    end
```

Comment:
It doesn't seem to be the same, is this intentional? https://github.com/theforeman/foreman/pull/10162/files#diff-05645e27e9ae6b798390ab2c548df6516a13db34d245e54eee0690c965bffbf4R21-R24

Reply:
It is supposed to create a hash where the `key` is the repository URL and the `value` is the corresponding GPG key. In this case, if you have the old parameters given, it just creates a hash with a single key value pair that is evaluated at a later point in time.

Reply:
I think I know what you mean. I messed up the params names but should be sorted out now.